packages/block-editor/src/components/block-preview/index.js

```diff
@@ -60,6 +60,7 @@ export function BlockPreview( {
 			...originalSettings,
 			focusMode: false, // Disable "Spotlight mode".
 			__unstableIsPreviewMode: true,
```

Comment:
Can we make this a getter and deprecate it or something?

Reply:
Maybe we can avoid passing all these __unstableIsPreviewMode configs and instead have a deprecated "property getter" in the default settings that fallbacks to isPreviewMode

Reply:
As much as I love that idea, it seems incredibly complex to do it right.

We're using the `SETTINGS_DEFAULTS` object in many places (`defaults.native`, the reducer, the Editor store), and the spread operator does not include the prototype, meaning the getter is not copied. Plus, overriding the settings in this file, for example, would also translate to the getter not being copied.

So, I think this duplication is better. Unfortunately, because we're not using the getter function, we cannot display the deprecation notice. However, a possible solution here would be to only add the getter function to the sub-registry settings, which is what this file does, and keep the reducer intact (and I believe we can even remove the `__unstableIsPreviewMode` property from it directly if we're feeling adventurous.)

Reply:
The native code is not that important. I think it's ok to not have the fallback there. 

So with that in mind, it seems like the "reducer" is the only place no, so it should be doable no?

Reply:
Maybe?

It'd help a lot if I knew how `<ExperimentalBlockEditorProvider />` works, so I think I'll check it out.

The main problem, as I have explained above, is that `...` doesn't include getters. This is true for the block editor provider cases, but there's also [this one](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/style-book/index.js#L94), which is not circling back to the reducer. On the other hand, I'm not sure why this `settings` override includes the preview mode since it's only making use of the `styles` property. Finally, there's also [this instance](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/provider/use-block-editor-settings.js#L283-L287) of spreading. I'm not sure at to what extent we need to preserve the property there, though.

Reply:
> We'll probably need to do the same for the editor settings reducer as it also exposes the preview mode flag. See useSpecificEditorSettings, its [usage](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/editor/index.js#L228) and [definition](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/editor/index.js#L67-L70). This is all uncharted water to me, so I might be wrong.

I'm not entirely sure that we need to do the same in "editor reducer" because I expect most consumers of this API to be in the block editor package. Editor provider just passes the value down. In theory, it's possible though but I'm not sure whether we should care too much here.

I say let's give it a try and see.

Reply:
That worked 👍🏻 I did some manual component testing that reproduced [this scenario](https://github.com/Automattic/jetpack/pull/39768/files#diff-7e61fb5ddc88a0cff10c780deef2738a28d45aff18b1a94894c95ed6b3188435R74-R80) (apart from the reducer automated testing), and I did see the deprecation notice in the console!